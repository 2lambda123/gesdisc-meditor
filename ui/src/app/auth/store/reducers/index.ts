import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as fromAuth from './auth';

export interface AuthState {
  status: fromAuth.State;
}


export const reducers = {
  status: fromAuth.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(
  selectAuthStatusState, 
  fromAuth.getUser
);