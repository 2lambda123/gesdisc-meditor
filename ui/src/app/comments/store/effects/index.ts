import { CommentsEffects } from './comments.effects';

export const effects: any[] = [CommentsEffects];

export * from './comments.effects';