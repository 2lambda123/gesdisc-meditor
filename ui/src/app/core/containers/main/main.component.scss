.app-container {
    background: #f0f0f0;
}

.app-version {
    font-size: 10px;
    position: absolute;
    bottom: 0;
    right: 10px;
}

med-toolbar {
    background: #fff;
    margin-bottom: 20px;
    padding: 0 40px;
    -webkit-box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);
    -moz-box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);
    box-shadow: 0 3px 5px rgba(57, 63, 72, 0.1);

    img {
        height: 64px;
    }
}

med-toolbar + div {
    background: #fff;
    margin: 0 auto;
    width: 1200px;
    padding: 15px !important;
}
