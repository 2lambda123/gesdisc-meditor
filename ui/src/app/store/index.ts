export { DocumentState } from './document/document.state';
export { ModelState } from './model/model.state';
export { NotificationState } from './notification/notification.state';
export { AuthState } from './auth/auth.state';
export { RouterState } from './router/router.state';
