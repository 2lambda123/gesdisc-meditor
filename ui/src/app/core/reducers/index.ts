import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
