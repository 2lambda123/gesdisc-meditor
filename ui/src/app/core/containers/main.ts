import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
import { Store, select } from '@ngrx/store';

import * as fromRoot from '../../reducers';
import * as ContentTypes from '../actions/content-type.actions';

@Component({
  selector: 'meditor-app',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <div fxFill fxLayout="column">
      <med-toolbar fxFlex="60px">
      </med-toolbar>
      <div fxFlex="100">
        <router-outlet></router-outlet>
      </div>
    </div>
  `
  ,
})
export class MainComponent implements OnInit {

  constructor(private store: Store<fromRoot.State>) {

  }

  ngOnInit() {
  	this.store.dispatch(new ContentTypes.LoadContentTypes());
  }

}
