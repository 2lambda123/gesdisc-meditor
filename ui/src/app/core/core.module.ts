import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
