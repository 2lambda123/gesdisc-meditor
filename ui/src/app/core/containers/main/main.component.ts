import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
import { environment } from 'environments/environment';

@Component({
  selector: 'meditor-app',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <div class="app-container" fxFill fxLayout="column" fxLayoutAlign="center none">
      <med-toolbar fxFlex="82px"></med-toolbar>
      <div fxFlex="90-100px" style="overflow: scroll; padding: 0 16px;">
        <router-outlet></router-outlet>
      </div>
      <div class="app-version">v{{version}}</div>
      <med-loading></med-loading>
      <med-notification></med-notification>
    </div>
  `,
  styleUrls: ['main.component.scss']
})
export class MainComponent implements OnInit {

  version: string = environment.VERSION

  constructor() {}

	ngOnInit() {}

}
