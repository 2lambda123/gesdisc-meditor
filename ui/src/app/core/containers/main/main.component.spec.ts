import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MainComponent } from './main.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { NgxsModule, Store, } from '@ngxs/store';
import { of } from 'rxjs';

describe('Main Page', () => {
	let fixture: ComponentFixture<MainComponent>;
	let component: MainComponent;
  let store: Store;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
        NgxsModule.forRoot(),
			],
			declarations: [
				MainComponent
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

    const initialState = {
			auth: {
				loggedIn: false
			}
		};

		fixture = TestBed.createComponent(MainComponent);
		component = fixture.componentInstance;

    Object.defineProperty(component, 'loggedIn$', { writable: true });
		component.loggedIn$ = of(initialState.auth.loggedIn);

    store = TestBed.get(Store);
		store.reset(initialState);

		spyOn(component, 'keepAlive').and.callThrough();

		fixture.detectChanges();
	});

	it('should compile main container and call keepAlive', () => {
		expect(fixture).toMatchSnapshot();
    component.ngOnInit();
		expect(component.keepAlive).toHaveBeenCalled();
	});

});
