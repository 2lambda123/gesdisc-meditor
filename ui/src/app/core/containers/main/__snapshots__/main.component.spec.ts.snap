// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page should compile 1`] = `
<meditor-app
  store={[Function Store]}
>
  <div
    _ngcontent-c0=""
    fxfill=""
    fxlayout="row"
  >
    <div
      _ngcontent-c0=""
      fxflex="5"
    />
    <div
      _ngcontent-c0=""
      fxflex="90"
    >
      <div
        _ngcontent-c0=""
        fxfill=""
        fxlayout="column"
        fxlayoutalign="center none"
      >
        <med-toolbar
          _ngcontent-c0=""
          fxflex="70px"
        />
        <div
          _ngcontent-c0=""
          fxflex="90-70px"
          style="overflow: scroll; padding: 0 16px;"
        >
          <router-outlet
            _ngcontent-c0=""
          />
        </div>
        <div
          _ngcontent-c0=""
          fxflex="10"
        />
      </div>
    </div>
    <div
      _ngcontent-c0=""
      fxflex="5"
    />
  </div>
</meditor-app>
`;
